src/utils/handle-request.ts

```diff
@@ -1,29 +1,19 @@
-export default async function handleRequest(request: Request, edgeCacheTtl: number) {
+export default async function handleRequest(request: Request, edgeCacheTtl: number): Promise<Response> {
+  // https://developers.cloudflare.com/workers/runtime-apis/request/#requestinitcfproperties
   const cfOptions = edgeCacheTtl > 0 ? {
-    // Requests proxied through Cloudflare are cached even without Workers according to a zone’s default
-    // or configured behavior. Setting Cloudflare cache rules to further customised the behavior
     cf: {
-      // NOTE: cf is Enterprise only feature
-      // This is equivalent to setting two Page Rules:
-      // - Edge Cache TTL and
-      // - Cache Level (to Cache Everything).
-      // Edge Cache TTL (Time to Live) specifies how long to cache a resource in the Cloudflare edge network
-      // The value must be zero or a positive number.
-      // A value of 0 indicates that the cache asset expires immediately.
-      // This option applies to GET and HEAD request methods only.
       cacheTtlByStatus: { '200-299': edgeCacheTtl, '404': 1, '500-599': 0 },
-      // The Cloudflare CDN does not cache HTML by default without setting cacheEverything to true.
       cacheEverything: true,
       cacheKey: request.url
     }
   } : {}
 
   //@ts-ignore
-  let response = await fetch(request, cfOptions);
+  let response = new Response(await fetch(request, cfOptions));
 
-  // Reconstruct the Response object to make its headers mutable.
-  response = new Response(response.body, response);
-  response.headers.set('edge-cache-ttl', `${edgeCacheTtl}`)
+  if (edgeCacheTtl > 0) { // Only set the header if edgeCacheTtl is set
+    response.headers.set('edge-cache-ttl', `${edgeCacheTtl}`);
```

Comment:
There's no point in forcing setting this to 0 if not defined, right?

Reply:
You're right, I'm just wondering whether it wouldn't be unexpected that everything is getting cached let's say for a day. It would definitely be needed to add `CircleCi --> Purge Cloudflare cache on deploy` for every project as well.

Reply:
agreed. Purging the cache after deploy should be in the standardized CircleCi config file :)